miden-lib/asm/wallet/basic.masm

```diff
@@ -0,0 +1,30 @@
+use.miden::sat::account
+use.miden::sat::tx
+use.miden::eoa::basic_auth
+
+#! Stack: [ASSET]
+#! Output: [ASSET']
+#!
+#! - ASSET' final asset in the account vault defined as follows:
+#!   - If ASSET is a non-fungible asset, then ASSET' is the same as ASSET.
+#!   - If ASSET is a fungible asset, then ASSET' is the total fungible asset in the account vault
+#!     after ASSET was added to it.
+export.receive_asset
+    exec.account::add_asset
+end
+
+#! Creates a note which sends a single asset to the specified recipient.
+#!
+#! Stack: [ASSET, tag, RECIPIENT]
+#! Output: [ptr]
+#!
+#! - ASSET is the non-fungible asset of interest.
+#! - tag is the tag to be included in the note.
+#! - RECIPIENT is the recipient of the note.
+#!
+#! - ptr is the pointer to the memory address at which the note is stored.
+export.send_asset
```

Comment:
A few nits similar to the comment before, but also a couple of other comments:

- I think returning pointer is fine, but we need to provide more info on what it is. This is pointer to where the note is written in the kernel memory. Thus, we can't access it directly from the note or account contexts - so, for now it is not very useful. But in the future we might provide additional kernel procedures which could manipulate the note after it was created.
- I think the stack inputs/outputs should look as follows:

```
Inputs:  [ASSET, RECIPIENT, tag, ...]
Outputs: [note_ptr, 0, 0, 0, 0, 0, 0, 0, 0, ...]
```

Reply:
ok, done. But we cannot use this pointer at any time in the future either, right? I return it for now, but to me it seems cleaner to return `0`s instead.

And it is unclear to me what a user would do with this note now and how it will be processed further. But let's have this for the future.